tst/com/amazon/corretto/crypto/provider/test/EvpKeyAgreementSpecificTest.java

```diff
@@ -129,9 +96,20 @@ private static void assertKeyEquals(String message, Key a, Key b) {
         assertArrayEquals(a.getEncoded(), b.getEncoded(), message);
     }
 
-    private static byte[] agree(byte[] privateKeyDer, byte[] publicKeyDer, int keyType)
+    private static byte[] agree(Key privateKeyRaw, Key publicKeyRaw)
       throws Throwable {
-        return sneakyInvokeExplicit(SPI_CLASS, "agree", null,
-            privateKeyDer, publicKeyDer, keyType, false);
+        KeyFactory kf = KeyFactory.getInstance("EC", NATIVE_PROVIDER);
+
+        Key privateKey = kf.translateKey(privateKeyRaw);
+        Key publicKey = kf.translateKey(publicKeyRaw);
+
+        // Horribly evil!
+        synchronized (privateKey) {
+            synchronized (publicKey) {
```

Comment:
We're handling raw pointers. All access to them must be synchronized (just in case). There still is the slight risk that we don't synchronize sufficiently, but it is a test case and I don't think it's a a real problem.

Also, lines 103 and 104 might return the keys unchanged if they are already of the proper type.